.github/workflows/test.yaml

```diff
@@ -28,46 +28,46 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@v4
+
       - name: Set up python ${{ inputs.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ inputs.python-version }}
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-      - name: Load cached venv
-        if: runner.os != 'Windows'
-        id: cached-poetry-dependencies
-        uses: actions/cache@v3
-        with:
-          path: .venv
-          key: v1-venv-${{ runner.os }}-${{ inputs.python-version }}-${{ inputs.pydantic-version }}-${{ hashFiles('**/poetry.lock') }}
-      - name: Load cached pip wheels
-        if: runner.os == 'Windows'
-        id: cached-pip-wheels
-        uses: actions/cache@v3
+
+      - uses: pdm-project/setup-pdm@v3
+        name: Set up PDM
         with:
-          path: ~/.cache
-          key: cache-${{ runner.os }}-${{ inputs.python-version }}-${{ inputs.pydantic-version }}-${{ hashFiles('**/poetry.lock') }}
+          python-version: ${{ inputs.python-version }}
+          allow-python-prereleases: true
+          cache: true
+          cache-dependency-path: |
+            ./pdm.lock
+
       - name: Install dependencies
-        run: poetry install --no-interaction --extras full
+        run: pdm install
```

Comment:
```suggestion
        run: pdm install -G:all
```

Reply:
As I understand it `pdm install` installs all test groups and prod. deps.